tests/components/home_plus_control/test_init.py

```diff
@@ -514,15 +595,24 @@ async def test_plant_api_timeout(
 
     # Register the mock responses
     aioclient_mock.get(
-        PLANT_URL,
+        PLANT_TOPOLOGY_BASE_URL,
         text=plant_data,
         exc=asyncio.TimeoutError,
     )
 
     # Load the entry
-    hass.data[DOMAIN] = {}
-    hass.config.components.add(DOMAIN)
-    await setup.async_setup_component(hass, "http", {})
+    assert await setup.async_setup_component(
+        hass,
+        "home_plus_control",
+        {
+            "home_plus_control": {
+                CONF_CLIENT_ID: CLIENT_ID,
+                CONF_CLIENT_SECRET: CLIENT_SECRET,
+                CONF_SUBSCRIPTION_KEY: SUBSCRIPTION_KEY,
+            },
+            "http": {},
+        },
+    )
     assert hass.http.app
     await hass.config_entries.async_add(mock_config_entry)
```

Comment:
Use this instead:
https://github.com/home-assistant/core/blob/50b5fc486082d101f0c40ca06bdc0522716af2c3/tests/common.py#L755